freqtrade/freqtradebot.py

```diff
@@ -512,39 +513,59 @@ def check_and_call_adjust_trade_position(self, trade: Trade):
-            if count_of_buys > self.strategy.max_entry_position_adjustment:
-                logger.debug(f"Max adjustment entries for {trade.pair} has been reached.")
-                return
-        else:
-            logger.debug("Max adjustment entries is set to unlimited.")
-        current_rate = self.exchange.get_rate(
-            trade.pair, side='entry', is_short=trade.is_short, refresh=True)
-        current_profit = trade.calc_profit_ratio(current_rate)
+        current_entry_rate, current_exit_rate = self.exchange.get_rates(
+            trade.pair, True, trade.is_short)
+
+        current_entry_profit = trade.calc_profit_ratio(current_entry_rate)
+        current_exit_profit = trade.calc_profit_ratio(current_exit_rate)
 
-        min_stake_amount = self.exchange.get_min_pair_stake_amount(trade.pair,
-                                                                   current_rate,
-                                                                   self.strategy.stoploss)
-        max_stake_amount = self.exchange.get_max_pair_stake_amount(trade.pair, current_rate)
+        min_entry_stake = self.exchange.get_min_pair_stake_amount(trade.pair,
+                                                                  current_entry_rate,
+                                                                  self.strategy.stoploss)
+        min_exit_stake = self.exchange.get_min_pair_stake_amount(trade.pair,
+                                                                 current_exit_rate,
+                                                                 self.strategy.stoploss)
+        max_entry_stake = self.exchange.get_max_pair_stake_amount(trade.pair, current_entry_rate)
+        max_exit_stake = self.exchange.get_max_pair_stake_amount(trade.pair, current_exit_rate)
         stake_available = self.wallets.get_available_stake_amount()
         logger.debug(f"Calling adjust_trade_position for pair {trade.pair}")
         stake_amount = strategy_safe_wrapper(self.strategy.adjust_trade_position,
                                              default_retval=None)(
-            trade=trade, current_time=datetime.now(timezone.utc), current_rate=current_rate,
-            current_profit=current_profit, min_stake=min_stake_amount,
-            max_stake=min(max_stake_amount, stake_available))
+            trade=trade, current_time=datetime.now(timezone.utc), current_rate=current_entry_rate,
+            current_profit=current_entry_profit, min_stake=min_entry_stake,
+            max_stake=min(max_entry_stake, stake_available),
+            current_entry_rate=current_entry_rate, current_exit_rate=current_exit_rate,
+            current_entry_profit=current_entry_profit, current_exit_profit=current_exit_profit,
+            min_entry_stake=min_entry_stake, min_exit_stake=min_exit_stake,
+            max_entry_stake=min(max_entry_stake, stake_available),
+            max_exit_stake=min(max_exit_stake, stake_available)
```

Comment:
This is a very dangerous change the way it is now.
Not because of the parameters, but because it's done completely inconsistently.
backtesting does not provide all parameters (so strategies relying on that will malfunction)- and none of the new parameters are in the documentation, nor in the function header (`interface.py`)

This method now has way too many parameters. It's not useful this way.

`max_exit_stake` is automatically the current trade amount  (so the parameter is irrelevant).
`min_exit_stake` should also be removed - it's simply `min_stake` - the difference in rate is too minimal to matter, so this only clutters the interface.

i also don't think the `current_*_profit` and `current_*_rate` parameters are necessary.
Depending on the config, their value will be identical - and having this many parameters will only confuse people using this.